### PR TITLE
feat: source setup page

### PR DESCRIPTION
**Pull request by joeclayallday**
This PR introduces a "Source Setup" page. It also adds the Source URL to the source list table.

If a user has no sources, they will be automatically redirected to the Source Setup page, and given instructions on how to set up a source for a selected app.

Screenshot 2024-04-19 at 5 15 06 PM

A user that already has at least one source configured can visit the Source Setup page by clicking the New Source button.

Screenshot 2024-04-22 at 11 51 23 AM

### CODE REVIEW

**Comment by joeclayallday** [comment]
Tightened the spacing up a bit

Screenshot 2024-04-19 at 6 01 06 PM

File: src/ui/pages/sources-setup.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+export function SourcesSetupPage() {
+  const { isLoading } = useQuery(fetchApps());
+  const apps = useSelector(selectAppsAsList);
+  const options: SelectOption[] = [
+    {
+      label: isLoading ? "Please wait..." : "Select an app",
+      value: "",
+      disabled: isLoading,
+    },
+    ...(apps || [])
```
Comment (neurosnap) [inline src/ui/pages/sources-setup.tsx:28]: `apps` should always be an empty list so this shouldn't be necessary.

**Review by neurosnap — COMMENTED** [review]
1 comment, otherwise, looks good!

File: src/ui/pages/sources-setup.tsx (hunk 1 above)
Comment (eabruzzese) [reply src/ui/pages/sources-setup.tsx]: Ah, this was to make TypeScript happy in a previous iteration, but it looks like it's no longer needed. Removed 👍

**Review by neurosnap — APPROVED** [review]

**Pull request merged**